### ISSUE TITLE
dplyr 1.0.4 introduces grouping bug

### ISSUE THREAD

**Issue by jntrcs** [tidyverse/dplyr#5769]
Dplyr is exhibiting strange, non-deterministic behavior on version 1.0.4. In the example below, running the first pipe causes the second one to error, whereas if you start with the second one it runs without error. From what I can tell, the error is because something is remembering the grouping of the first pipe, and when the second pipe has a different number of groups, it causes an error. This is only present when I use both a non-across step in mutate (using the `n()` function) and an `across` step.

I have tested this on dplyr 1.0.3 and do not experience the same error.

---

```r
library(dplyr)

mtcars %>% filter(cyl>=6) %>% 
  group_by(cyl) %>%
  mutate(num=n(),
         across(carb, ~.x*10))

mtcars %>% filter(cyl<6) %>% 
  group_by(cyl) %>%
  mutate(num=n(),
         across(carb, ~.x*10))
```

**Comment by meztez** [comment]
Related https://github.com/tidyverse/dplyr/pull/5765?

**Comment by romainfrancois** [comment]
This was indeed fixed by #5765. 

``` r
library(dplyr, warn.conflicts = FALSE)

mtcars %>% 
  filter(cyl>=6) %>% 
  group_by(cyl) %>%
  mutate(num=n(), across(carb, ~.x*10))
#> # A tibble: 21 x 12
#> # Groups:   cyl [2]
#>      mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb   num
#>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
#>  1  21       6  160    110  3.9   2.62  16.5     0     1     4    40     7
#>  2  21       6  160    110  3.9   2.88  17.0     0     1     4    40     7
#>  3  21.4     6  258    110  3.08  3.22  19.4     1     0     3    10     7
#>  4  18.7     8  360    175  3.15  3.44  17.0     0     0     3    20    14
#>  5  18.1     6  225    105  2.76  3.46  20.2     1     0     3    10     7
#>  6  14.3     8  360    245  3.21  3.57  15.8     0     0     3    40    14
#>  7  19.2     6  168.   123  3.92  3.44  18.3     1     0     4    40     7
#>  8  17.8     6  168.   123  3.92  3.44  18.9     1     0     4    40     7
#>  9  16.4     8  276.   180  3.07  4.07  17.4     0     0     3    30    14
#> 10  17.3     8  276.   180  3.07  3.73  17.6     0     0     3    30    14
#> # … with 11 more rows

mtcars %>% 
  filter(cyl<6) %>% 
  group_by(cyl) %>%
  mutate(num=n(), across(carb, ~.x*10))
#> # A tibble: 11 x 12
#> # Groups:   cyl [1]
#>      mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb   num
#>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
#>  1  22.8     4 108      93  3.85  2.32  18.6     1     1     4    10    11
#>  2  24.4     4 147.     62  3.69  3.19  20       1     0     4    20    11
#>  3  22.8     4 141.     95  3.92  3.15  22.9     1     0     4    20    11
#>  4  32.4     4  78.7    66  4.08  2.2   19.5     1     1     4    10    11
#>  5  30.4     4  75.7    52  4.93  1.62  18.5     1     1     4    20    11
#>  6  33.9     4  71.1    65  4.22  1.84  19.9     1     1     4    10    11
#>  7  21.5     4 120.     97  3.7   2.46  20.0     1     0     3    10    11
#>  8  27.3     4  79      66  4.08  1.94  18.9     1     1     4    10    11
#>  9  26       4 120.     91  4.43  2.14  16.7     0     1     5    20    11
#> 10  30.4     4  95.1   113  3.77  1.51  16.9     1     1     5    20    11
#> 11  21.4     4 121     109  4.11  2.78  18.6     1     1     4    20    11
```

<sup>Created on 2021-02-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
